File: app/src/main/java/com/borisruzanov/russianwives/mvp/model/interactor/MyProfileInteractor.java
package com.borisruzanov.russianwives.mvp.model.interactor;

import com.borisruzanov.russianwives.mvp.model.repository.FirebaseRepository;
import com.borisruzanov.russianwives.utils.ActionsCountInfoCallback;
import com.borisruzanov.russianwives.utils.UserCallback;

public class MyProfileInteractor {

    private FirebaseRepository repository;

    public MyProfileInteractor(FirebaseRepository repository) {
        this.repository = repository;
    }

    public void getAllCurrentUserInfo(UserCallback callback){
        repository.getAllCurrentUserInfo(callback);
    }

    public void getActionsCountInfo(ActionsCountInfoCallback callback){
        repository.getActionsCountInfo(callback);
    }


}

File: app/src/main/java/com/borisruzanov/russianwives/mvp/model/interactor/myprofile/MyProfileInteractor.kt
package com.borisruzanov.russianwives.mvp.model.interactor.myprofile

import com.borisruzanov.russianwives.mvp.model.repository.user.UserRepository
import com.borisruzanov.russianwives.utils.ActionsCountInfoCallback
import com.borisruzanov.russianwives.utils.UserCallback
import javax.inject.Inject

class MyProfileInteractor @Inject constructor(private val repository: UserRepository) {

    fun getAllCurrentUserInfo(callback: UserCallback) {
        repository.getAllCurrentUserInfo(callback)
    }

    fun getActionsCountInfo(callback: ActionsCountInfoCallback) {
        repository.getActionsCountInfo(callback)
    }


}
